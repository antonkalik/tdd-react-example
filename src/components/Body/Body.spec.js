import { render, screen } from "@testing-library/react";
import { Body } from "./index";

describe("Body", () => {
  it("should render body", () => {
    render(<Body />);
    expect(screen.getByTestId("body")).toMatchSnapshot();
  });

  it("should render message with no cards", () => {
    render(<Body />);
    const element = screen.getByTestId("no-cards-message");
    expect(element).toBeInTheDocument();
  });

  it("should render title", () => {
    const { container } = render(<Body />);
    const title = container.querySelector("h3");
    expect(title).toHaveTextContent("Cards");
  });

  it("should render one card", () => {
    render(
      <Body
        cards={[
          {
            id: 1,
            title: "Card 1",
          },
        ]}
      />
    );
    const element = screen.getByTestId("card");
    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent("Card 1");
  });
});
