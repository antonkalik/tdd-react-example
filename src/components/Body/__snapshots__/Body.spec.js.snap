// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render body 1`] = `
<div
  data-testid="body"
>
  <h3>
    Cards
  </h3>
  <div
    data-testid="cards"
  >
    <div
      data-testid="no-cards-message"
    >
      No cards
    </div>
  </div>
</div>
`;
